pkg/storage/runtime.go

```diff
@@ -0,0 +1,418 @@
+	// StopContainer attempts to unmount a container's root filesystem,
+	// freeing up any kernel resources which may be limited.
+	StopContainer(idOrName string) error
+
+	// GetWorkDir returns the path of a nonvolatile directory on the
+	// filesystem (somewhere under the Store's Root directory) which can be
+	// used to store arbitrary data that is specific to the container.  It
+	// will be removed automatically when the container is deleted.
+	GetWorkDir(id string) (string, error)
+	// GetRunDir returns the path of a volatile directory (does not survive
+	// the host rebooting, somewhere under the Store's RunRoot directory)
+	// on the filesystem which can be used to store arbitrary data that is
+	// specific to the container.  It will be removed automatically when
+	// the container is deleted.
+	GetRunDir(id string) (string, error)
+}
+
+// RuntimeContainerMetadata is the structure that we encode as JSON and store
+// in the metadata field of storage.Container objects.  It is used for
+// specifying attributes of pod sandboxes and containers when they are being
+// created, and allows a container's MountLabel, and possibly other values, to
+// be modified in one read/write cycle via calls to
+// RuntimeServer.GetContainerMetadata, RuntimeContainerMetadata.SetMountLabel,
+// and RuntimeServer.SetContainerMetadata.
+type RuntimeContainerMetadata struct {
+	// Pod is true if this is the pod's infrastructure container.
+	Pod bool `json:"pod,omitempty"` // Applicable to both PodSandboxes and Containers
+	// The pod's name and ID, kept for use by upper layers in determining
+	// which containers belong to which pods.
+	PodName string `json:"pod-name"` // Applicable to both PodSandboxes and Containers, mandatory
+	PodID   string `json:"pod-id"`   // Applicable to both PodSandboxes and Containers, mandatory
+	// The provided name and the ID of the image that was used to
+	// instantiate the container.
+	ImageName string `json:"image-name"` // Applicable to both PodSandboxes and Containers
+	ImageID   string `json:"image-id"`   // Applicable to both PodSandboxes and Containers
+	// The container's name, which for an infrastructure container is usually PodName + "-infra".
+	ContainerName string `json:"name"` // Applicable to both PodSandboxes and Containers, mandatory
+	// The name as originally specified in PodSandbox or Container CRI metadata.
+	MetadataName string `json:"metadata-name"`        // Applicable to both PodSandboxes and Containers, mandatory
+	UID          string `json:"uid,omitempty"`        // Only applicable to pods
+	Namespace    string `json:"namespace,omitempty"`  // Only applicable to pods
+	Attempt      uint32 `json:"attempt,omitempty"`    // Applicable to both PodSandboxes and Containers
+	CreatedAt    int64  `json:"created-at"`           // Applicable to both PodSandboxes and Containers
+	MountLabel   string `json:"mountlabel,omitempty"` // Applicable to both PodSandboxes and Containers
+}
```

Comment:
I suppose we could, if we have one.  It's already being kept with the image.  Is ocid looking to start pulling defaults from it?